skimage/draw/draw.py

```diff
@@ -693,3 +693,57 @@ def bezier_curve(r0, c0, r1, c1, r2, c2, weight, shape=None):
            [0, 0, 0, 0, 0, 0, 0, 0, 0, 0]], dtype=uint8)
     """
     return _bezier_curve(r0, c0, r1, c1, r2, c2, weight, shape)
+
+
+def rectangle(origin, extent, shape=None):
+    """Generate coordinates of pixels within an nd-rectangle.
+
+    Parameters
+    ----------
+    origin : tuple
+        origin of the rectangle, (page,row,column,channel)
+    extent : tuple
+        extent (num_pages,num_rows,num_cols,num_chanels..) of the rectangle.
```

Comment:
Looking at other functions in `draw`, sometimes we use center coordinates and extent, sometimes (e.g. line), we use (from, to). I think for rectangle I would prefer the latter, but not sure. So, rather than (origin, extent) as the arguments, I would say (top, bottom) or (start, end). I'm not set on this, but would like to discuss it. My rationale is that I suspect that is the most common use case.